src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const groupKey = item[key] as string;
```

Comment:
The assumption that `item[key]` is always a string may not hold true for all types of `T`. If `key` can be of a type other than string, this cast might lead to runtime errors. Consider using `String(item[key])` to ensure the key is a string.